cost/forecasting/straight_line_forecast/straight_line_forecast.pt

```diff
@@ -0,0 +1,380 @@
+      body_fields: {
+        "dimensions": ["billing_center_id"],
+        "granularity": "month",
+        "metrics": [cost_metric[param_cost_metric]],
+        "billing_center_ids": billing_center_ids,
+        "start_at": start_at,
+        "end_at": end_at
+      },
+      headers: {
+        "Api-Version": "1.0",
+        "User-Agent": "RS Policies",
+      }
+    }
+  EOS
+end
+
+script "js_make_data_package", type: "javascript" do
+  parameters "param_cost_metric", "param_forecasted_months", "ds_costs"
+  result "results"
+  code <<-EOS
+  var results = [];
+
+  var cost_metric = {
+    "Unamortized Unblended":"cost_nonamortized_unblended_adj",
+    "Amortized Unblended":"cost_amortized_unblended_adj",
+    "Unamortized Blended": "cost_nonamortized_blended_adj",
+    "Amortized Blended":"cost_amortized_blended_adj"
+  }
+
+  var mapped_costs = _.groupBy(ds_costs, function (value) {
+    return value.start_date;
+  });
+  var unsorted_results = _.map(mapped_costs, function (group) {
+    var counter = 0;
+    for (var i = 0; i < group.length; i++) {
+      var item = group[i];
+      if (item[cost_metric[param_cost_metric]] == 0) {
+        counter = counter + 1;
+      } else {
+        break;
+      }
+    }
+
+    var arr_new_sum = _.pluck(group, cost_metric[param_cost_metric]);
+
+    var summed = _.reduce(arr_new_sum, function (memo, num) { return memo + num; }, 0);
+    return {
+      start_date: group[0].start_date,
+      cost: summed,
+    };
+  });
+  var costs = [];
+  _.each(unsorted_results, function (item) {
+    costs.push({
+      start_date: item.start_date,
+      cost: item.cost,
+    });
+  });
+
+  var sorted_costs = _.sortBy(costs, function (o) { var dt = new Date(o.start_date); return -dt; });
+  console.log(sorted_costs);
+  results.push({
+    costs: sorted_costs,
+    forecasted_months: param_forecasted_months,
+  });
+EOS
+end
+
+script "js_forecast", type: "javascript" do
+  parameters "ds_make_data_package"
+  result "report"
+  code <<-EOS
+  var forecasted_costs = [];
+  function getFormattedDailyDate(date) {
+    var year = date.getFullYear();
+    var month = (1 + date.getMonth()).toString();
+    month = month.length > 1 ? month : '0' + month;
+    return year + '-' + month;
+  }
+  function getFutureMonth( date, month_counter ) {
+    date.setMonth(date.getMonth() + month_counter);
+    return date;
+  }
+
+  var costs = ds_make_data_package[0].costs;
+  var forecasted_months = ds_make_data_package[0].forecasted_months;
+  var end_cost = costs[0].cost;
+  var start_cost = costs[costs.length - 1].cost;
+  var percent_difference = ((end_cost / start_cost) - 1);
+  var monthly_percent_difference = (percent_difference / costs.length);
+  var forecasted_cost = costs[0].cost;
+  var chartDataArray = [];
+  var XAxisArray = [];
+  var monthNames = ['January', 'February', 'March', 'April', 'May', 'June',
+    'July', 'August', 'September', 'October', 'November', 'December',
```

Comment:
Got a comma at the end of the array after 'December', probably not a code breaker though.

Reply:
eslint made me :)

Reply:
https://eslint.org/docs/rules/comma-dangle